pandas/io/json/_normalize.py

```diff
@@ -267,6 +267,11 @@ def _pull_records(js: Dict[str, Any], spec: Union[List, str]) -> List:
         data = [data]
 
     if record_path is None:
+        if np.ndim(data) == 0:
```

Comment:
What about inserting a call to next(data) to skip past the first element?

It's always good to be as lazy as possible but tbf I don't know how much that matters here

Reply:
@arw2019 wdym with skipping past the first line with "next(data)"? Cause we do not want to skip the first element.

Below I've written down my understanding and solution to the problem. But lmk if you see this differently!

The problem here is that in the old situation the first line of a generator was consumed and not part of the output dataframe. This was caused by this line: https://github.com/pandas-dev/pandas/blob/master/pandas/io/json/_normalize.py#L270.

By consuming the whole generator and putting it in a list we don't have that issue anymore

Reply:
Okay right your solution looks good

(There's actually already a TODO there about expanding the generator with nested records, as well)

Reply:
Ah great. Wrt to the TODO, which one do you mean? I see this one: `TODO: handle record value which are lists, at least error reasonably`, but IMO that would justify its own PR since it is related to a different issue

Reply:
Agreed, that TODO is beyond the scope of this PR

Reply:
you can just put a list(data) on L264 (you could even limit this to a Iterable type if you want)

Reply:
this entire function needs to be split up into module level functions and cleaned up. I believe we have several open issues about this (but orthogonal to this PR).

Reply:
Put `list(data)` on L264, but limited it to  type `Iterator` and not `Iterable` since we use different conversion logic for dicts

Wrt to the splitting of the modules. I will add it to my list, but first I want to address some other PRs for dtypes.